src/files/castleclientserver.pas

```diff
@@ -0,0 +1,737 @@
+{
+  Copyright 2018 Benedikt Magnus.
+
+  This file is part of "Castle Game Engine".
+
+  "Castle Game Engine" is free software; see the file COPYING.txt,
+  included in this distribution, for details about the copyright.
+
+  "Castle Game Engine" is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+
+  ----------------------------------------------------------------------------
+}
+
```

Comment:
Please place a short documentation before the unit name.

We use this convention with all Castle Game Engine units, so that they look nice when viewed on a generated units list on https://castle-engine.io/apidoc/html/AllUnits.html . The 1st sentence (up to the 1st dot) is special (we use PasDoc with auto-abstract and auto-linking).

I propose this:

```
{ Client and server for network communication using TCP (TCastleTCPClient, TCastleTCPServer). 
  Use this in your games for easy client/server communication. 
  On Android,it requires adding a service client_server (see https://github.com/castle-engine/castle-engine/wiki/Android-Project-Services-Integrated-with-Castle-Game-Engine). 
  On other platforms, it requires having Indy (http://www.indyproject.org/ ) available 
  (note that Indy is not distributed together with Castle Game Engine, 
  you need to download it yourself, and point the compiler to find Indy sources). }
unit CastleClientServer;
```